extensions/rvdatabinding/src/main/java/com/google/android/agera/rvdatabinding/DataBindingRepositoryPresenterCompiler.java

```diff
@@ -36,18 +37,24 @@
 import com.google.android.agera.Function;
 import com.google.android.agera.Result;
 import com.google.android.agera.rvadapter.RepositoryPresenter;
+import com.google.android.agera.rvadapter.RepositoryPresenterCompilerStates.RPCollectionCompile;
 import com.google.android.agera.rvadapter.RepositoryPresenterCompilerStates.RPItemCompile;
 import com.google.android.agera.rvadapter.RepositoryPresenterCompilerStates.RPLayout;
+import com.google.android.agera.rvadapter.RepositoryPresenterCompilerStates.RPTypedCollectionCompile;
 import com.google.android.agera.rvdatabinding.DataBindingRepositoryPresenterCompilerStates.DBRPMain;
 import java.lang.ref.WeakReference;
 import java.util.List;
 
 @SuppressWarnings("unchecked")
-final class DataBindingRepositoryPresenterCompiler implements DBRPMain, RPLayout {
+final class DataBindingRepositoryPresenterCompiler
+    implements DBRPMain, RPLayout, RPTypedCollectionCompile {
+  @NonNull
+  private static final Function NO_ID = staticFunction(-1);
```

Comment:
consider using 0 because [0 is not a valid resource ID](https://developer.android.com/reference/android/content/res/Resources.html#getIdentifier%28java.lang.String,%20java.lang.String,%20java.lang.String%29). also consider using View.NO_ID (-1) because it's publicly defined. This means you should check for both -1 and 0 (but allow any other value, positive or negative) further down in your code.